src/Discord.Net.Commands/Extensions/CommandServiceExtensions.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Discord.Commands
+{
+    public static class CommandServiceExtensions
+    {
+        public static async Task<IReadOnlyCollection<CommandInfo>> GetExecutableCommandsAsync(this IEnumerable<CommandInfo> commands, ICommandContext context, IServiceProvider provider)
+        {
+            var executableCommands = new List<CommandInfo>();
```

Comment:
Use `yield return` here instead of constructing a list

Reply:
Can't yield return in async methods.

Reply:
Quick solution: have an internal function which does the main body, returning an IEnumerable<Task<CommandInfo>>, and then use Task.WhenAll to return an array for them all.

Reply:
Hmmpfh. I suppose so. In that case, I suppose there's no point jumping through hoops just to use yield return syntax.

Reply:
Task.WhenAll's probably a good idea though.